### PR TITLE
[graphql]: Update to 0.12.x

### PR DESCRIPTION
**Pull request by dyst5422**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/graphql/graphql-js/releases/tag/v0.12.0>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/graphql/language/ast.d.ts [hunk 1]
```diff
@@ -431,27 +505,82 @@ export interface EnumTypeDefinitionNode {
...
     fields: InputValueDefinitionNode[];
 }
 
-export interface TypeExtensionDefinitionNode {
-    kind: 'TypeExtensionDefinition';
-    loc?: Location;
-    definition: ObjectTypeDefinitionNode;
-}
+export type TypeExtensionNode =
+  | ScalarTypeExtensionNode
```
Comment (IvanGoncharov) [inline types/graphql/language/ast.d.ts]: @dyst5422 Great PR 👍 

One problem I see is the inconsistency in indent (4vs2). But I think it should be handled in separate PR. Ideally, someone would copy [.prettierrc](https://github.com/graphql/graphql-js/blob/master/.prettierrc) from `graphql-js` and run it on entire `types/graphql/` folder.

**Review by IvanGoncharov — APPROVED** [review]

File: types/graphql/language/ast.d.ts (hunk 1 above)
Comment (dyst5422) [reply types/graphql/language/ast.d.ts]: Fair enough. I didn't want to run prettier since it would use my own settings.
